src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,158 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(head, value);
+        if (tail == null) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            head.next = newNode;
+            head = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Objects.checkIndex(index,size + 1);
+        Node<T> nodeForeAdd = new Node<>(value);
+        if (tail == null) {
+            tail = head = nodeForeAdd;
+        } else if (index == 0) {
+            nodeForeAdd.next = tail;
+            tail = nodeForeAdd;
```

Comment:
why the `tail` is chanded here?  when we add element to `0` index that means that we add a new `head`.